playground/nuxt.config.ts

```diff
@@ -24,6 +24,7 @@ export default defineNuxtConfig({
     rateLimiter: {
       tokensPerInterval: 10,
       interval: 10000
-    }
+    },
+    runtimeHooks: true
```

Comment:
question: I wonder if this should actually be enabled by default. I think that majority of the users will just use the nuxt.config.ts config or route rules while this runtime configuration is a bit of an edge case (at least IMO).

With these kind of configuration, I usually mark them as optional and disabled by default (to not ship unused code to other users).

So, if you want to have a runtime configuration, just set this config value `security.runtimeHooks = false` and it will work for you.

WDYT?

Reply:
oh i didn't remove this since the removal of the `runtimeHooks` option

@harlan-zw think we don't really need this option to always have a runtime configuration of headers. So should we set back `runtimeHooks` in the options ?

If we set back `runtimeHooks` option, i think it should be default to false. Then if true, we'll add the nitro plugin

Reply:
I personally would recommend to add an option and make it disabled by default to not add additional code for users that may not use it.

So add an option but make it disabled by default and make a note in the docs that in order to use this feature, set this config option to true and then, call the hook (like you have explained in the docs)

I think this delivers the functionality to the user that may need it without forcing every user of the module to have some dead code.